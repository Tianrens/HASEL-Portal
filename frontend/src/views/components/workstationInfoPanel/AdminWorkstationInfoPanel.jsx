import React, { useState } from 'react';

import { Button, ListItemIcon, ListItemText, Menu, MenuItem, Paper } from '@material-ui/core';
import { Link } from 'react-router-dom';
import Icon from '@material-ui/core/Icon';
import WorkstationInfoPanel from './WorkstationInfoPanel';
import styles from './AdminWorkstationInfoPanel.module.scss';

export default function AdminWorkstationInfoPanel({ workstationData }) {
    const menuItems = [
        {
            icon: <Icon>add</Icon>,
            title: 'Create Booking',
            link: '/bookings/new',
        },
        {
            icon: <Icon>view_list</Icon>,
            title: 'View Bookings',
            link: `/workstations/${workstationData._id}/bookings`,
        },
        {
            icon: <Icon>edit</Icon>,
            title: 'Edit Workstation',
            link: `/workstations/${workstationData._id}`,
        },
    ];
    const [anchorEl, setAnchorEl] = useState(null);
    const open = Boolean(anchorEl);
    const handleClick = (event) => {
        setAnchorEl(event.currentTarget);
    };
    const handleClose = () => {
        setAnchorEl(null);
    };

    return (
        <WorkstationInfoPanel workstationData={workstationData}>
            <div className={styles.adminButtonWrapper}>
                <Button className={styles.menuButton} onClick={handleClick}>
                    <Icon>more_horiz</Icon>
                </Button>
                <Paper>
                    <Menu anchorEl={anchorEl} open={open} onClose={handleClose}>
                        {menuItems.map((item) => (
                            <MenuItem
                                key={item.title}
                                component={Link}
                                to={{
                                    pathname: item.link,
                                    state: { workstationId: workstationData._id },
                                }}
                            >
                                <ListItemIcon>{item.icon}</ListItemIcon>
                                <ListItemText>{item.title}</ListItemText>
                            </MenuItem>
                        ))}
                    </Menu>
                </Paper>
            </div>
        </WorkstationInfoPanel>
    );
}
