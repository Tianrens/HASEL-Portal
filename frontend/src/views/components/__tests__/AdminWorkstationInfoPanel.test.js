import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider } from '@material-ui/core';
import AdminWorkstationInfoPanel from '../workstationInfoPanel/AdminWorkstationInfoPanel';

Object.defineProperty(global.self, 'crypto', {
    value: {
        // eslint-disable-next-line no-unused-vars
        getRandomValues: (arr) => null,
    },
});

const workstationData = [
    {
        _id: 0,
        name: 'Zeus Suis',
        cpuDescription: 'Super fast CPU',
        gpuDescription: 'Cool GPU',
        ramDescription: 'Ram ranch',
    },
];

test('BookingForm renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <AdminWorkstationInfoPanel workstationData={workstationData} />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
