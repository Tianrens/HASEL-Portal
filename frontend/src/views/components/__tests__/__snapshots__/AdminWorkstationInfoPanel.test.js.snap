// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingForm renders properly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 wrapper css-1ps6pg7-MuiPaper-root"
>
  <div
    className="infoContent"
  >
    <div
      className="workstationName"
    >
      <div
        className="wrapper large "
      >
        <img
          className="icon "
          src="workstation.svg"
        />
        <div
          className="spacer"
        />
        <div>
          <b />
           
        </div>
      </div>
    </div>
    <div
      className="infoFields"
    >
      <div
        className="wrapper undefined responsiveWrapper"
      >
        <img
          alt="CPU"
          className="icon responsiveIcon"
          src="cpu.svg"
        />
        <div
          className="spacer"
        />
        <div>
          <b>
            CPU
          </b>
           
        </div>
      </div>
      <div
        className="wrapper undefined responsiveWrapper"
      >
        <img
          alt="RAM"
          className="icon responsiveIcon"
          src="ram.svg"
        />
        <div
          className="spacer"
        />
        <div>
          <b>
            RAM
          </b>
           
        </div>
      </div>
      <div
        className="wrapper undefined responsiveWrapper"
      >
        <img
          alt="GPU"
          className="icon responsiveIcon"
          src="gpu.svg"
        />
        <div
          className="spacer"
        />
        <div>
          <b>
            GPU
          </b>
           
          : undefinedx undefined 
        </div>
      </div>
    </div>
    <div
      className="adminButtonWrapper"
    >
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root menuButton css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          aria-hidden={true}
          className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
        >
          more_horiz
        </span>
      </button>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
      />
    </div>
  </div>
  <div
    className="ganttLoadingWrapper"
  >
    <div
      className="ganttWrapper"
    >
      <div
        className="container"
      >
        <div
          className="chase"
        >
          <div
            className="chaseDot"
          />
          <div
            className="chaseDot"
          />
          <div
            className="chaseDot"
          />
          <div
            className="chaseDot"
          />
          <div
            className="chaseDot"
          />
          <div
            className="chaseDot"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
